docs/source/evaluation_scripts.md

```diff
@@ -129,3 +129,205 @@ Here are the steps to configure remote evaluation:
+                return list(range(self.env.action_space.n))
+
+            def next_score(self):
+                self.score += 1
+        ```
+
+        The attribute `self.env` here is the environment, `self.feedback` is the observations from the sensors of the agent after performing an action and `self.score` stores the score at each step.
+
+        The `next_score` method shows how the score can be updated and shown towards the end of the evaluation and the `get_action_space` gets the list of possible actions. You can add custom methods and attributes which help in interaction with the environment.
+
+    2. _Edit the Environment service_: This service is hosted on the GRPC server in order to get actions in form of messages from the agent container. Modify the lines shown [here](https://github.com/Cloud-CV/EvalAI-Starters/blob/8338085c6335487332f5b57cf7182201b8499aad/code_upload_challenge_evaluation/environment/environment.py#L35-L65):
+
+        ```python
+        class Environment(evaluation_pb2_grpc.EnvironmentServicer):
+            def __init__(self, challenge_pk, phase_pk, submission_pk, server):
+                self.challenge_pk = challenge_pk
+                self.phase_pk = phase_pk
+                self.submission_pk = submission_pk
+                self.server = server
+
+            def get_action_space(self, request, context):
+                message = pack_for_grpc(env.get_action_space())
+                return evaluation_pb2.Package(SerializedEntity=message)
+
+            def act_on_environment(self, request, context):
+                global EVALUATION_COMPLETED
+                if not env.feedback or not env.feedback[2]:
+                    action = unpack_for_grpc(request.SerializedEntity)
+                    env.next_score()
+                    env.feedback = env.env.step(action)
+                if env.feedback[2]:
+                    if not LOCAL_EVALUATION:
+                        update_submission_result(
+                            env, self.challenge_pk, self.phase_pk, self.submission_pk
+                        )
+                    else:
+                        print("Final Score: {0}".format(env.score))
+                        print("Stopping Evaluation!")
+                        EVALUATION_COMPLETED = True
+                return evaluation_pb2.Package(
+                    SerializedEntity=pack_for_grpc(
+                        {"feedback": env.feedback, "current_score": env.score,}
+                    )
+                )
+        ```
+
+        You can modify the relevant parts of the environment service in order to make it work for your case. For example, `feedback` can totally depend on the results obtained from the `env.env` after performing an action, which in-turn depends on the sensors expected from the agent. In our example, `feedback[2]` is the stop signal sensor. The `unpack_for_grpc` and `pack_for_grpc` are methods to receive and send messages with the agent.
```

Comment:
Can you just describe this in bullets in this format:

method_name: <1-2 line description>

Reply:
And also explain that hosts need to serialize and deserialize the response/request to act in the environment

Reply:
Added and explained.